### PR TITLE
Fix bugs with tasks

### LINKED ISSUE #161: Double clicking plans won't change current plan

### LINKED ISSUE #174: Updating problem bug
When updating the problem in the problem list, the same problem does not get updated in the plan

<hr><sub>[original: dragontho/ped#1]<br/>
</sub>

### LINKED ISSUE #175: Purpose of plan vs problem list
Not sure if the plan created is merely an organisation of the problem list or a copy of all tasks that need to be done at the point of creating the plan. Please confirm.

<hr><sub>[original: dragontho/ped#2]<br/>
</sub>

### LINKED ISSUE #193: Clear doesnt clear the entries on PlanTab
<hr><sub>[original: honhaochen/ped#7]<br/>
</sub>

### LINKED ISSUE #194: Mindful of your terminologies
Some question to consider is:

1. What is the different between Task and Problems?

2. Does user need to know himself that "add" only adds a problem, while "addtag", "addtask", "addplan" does other things? Can consider changing to "addprob"?

<hr><sub>[original: honhaochen/ped#8]<br/>
</sub>

### LINKED ISSUE #196: "CurrentPlan" Tab
Anyway to display the "CurrentPlan" tab by clicking the plans in PlanTab?

Right now it just switches automatically on add and edit

<hr><sub>[original: honhaochen/ped#10]<br/>
</sub>

### LINKED ISSUE #202: Hide the pie chart legend when pie chart is hidden
image

### PR DESCRIPTION
**Pull request by jiayushe**
- Fix #161 and Fix #196 
- Fix #174 and Fix #175 
- Fix #193 
- Fix #194 
- Fix #202

### CODE REVIEW

File: src/main/java/seedu/algobase/logic/commands/task/CopyTaskCommand.java [hunk 1]
```diff
@@ -82,7 +82,7 @@ public CommandResult execute(Model model, CommandHistory history) throws Command
         }
         taskSetTo.add(taskToCopy);
 
-        Plan updatedPlanTo = Plan.updateTasks(planTo, taskSetTo);
+        Plan updatedPlanTo = planTo.updateTasks(taskSetTo);
```
Comment (le0tan) [inline src/main/java/seedu/algobase/logic/commands/task/CopyTaskCommand.java]: `planTo` is probably not a good variable name.

File: src/main/java/seedu/algobase/model/AlgoBase.java [hunk 2]
```diff
@@ -157,10 +154,19 @@ public boolean checkIsProblemUsed(Problem problem) {
      * Removes the given problem from all plans
      * @param problem the problem to be removed
      */
-    public void removeProblemFromAllPlans(Problem problem) {
+    void removeProblemFromAllPlans(Problem problem) {
```
Comment (le0tan) [inline src/main/java/seedu/algobase/model/AlgoBase.java:158]: Making these methods package private might not be necessary, can you justify?

File: src/main/java/seedu/algobase/ui/CommandBox.java [hunk 3]
```diff
@@ -27,7 +27,7 @@
     private TextField commandTextField;
 
 
-    public CommandBox(CommandExecutor commandExecutor, List<String> history) {
+    CommandBox(CommandExecutor commandExecutor, List<String> history) {
```
Comment (le0tan) [inline src/main/java/seedu/algobase/ui/CommandBox.java]: Unnecessary change.

**Review by le0tan — CHANGES_REQUESTED** [review]
You changed a lot of access modifiers, are they necessary? Because AB3 has lots of public methods that IDEA will prompt you to change to private or package private methods...

File: src/main/java/seedu/algobase/logic/commands/task/CopyTaskCommand.java (hunk 1 above)
Comment (jiayushe) [reply src/main/java/seedu/algobase/logic/commands/task/CopyTaskCommand.java]: Hmm do you have any suggestions?

File: src/main/java/seedu/algobase/model/AlgoBase.java (hunk 2 above)
Comment (jiayushe) [reply src/main/java/seedu/algobase/model/AlgoBase.java:158]: They are not needed outside model package, so no need to expose them as public. They should be called through the model interface.

**Review by le0tan — APPROVED** [review]

**Pull request merged**